awx_collection/plugins/module_utils/controller_api.py

```
@@ -512,121 +583,155 @@ def make_request(self, method, endpoint, *args, **kwargs):
+        # ----------------------------------------------------------------
+        # Retry loop — wraps only the session.open() + HTTPError handling
+        # Everything above (auth, URL building) happens once before the loop
+        # ----------------------------------------------------------------
+        max_retries = self.max_retries
+        backoff_factor = self.retry_backoff_factor
+        last_response = None
+
+        for attempt in range(max_retries + 1):  # attempt 0 = first try
+
+            if attempt > 0:
+                sleep_time = (backoff_factor ** (attempt - 1)) * (0.5 + random.random())
```

Ruff (0.15.10)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)